src/vehicle.cpp

```diff
@@ -5393,7 +5393,7 @@ void vehicle::open_or_close(int part_index, bool opening)
       int ydiff = parts[next_index].mount_dy - parts[part_index].mount_dy;
       if((xdiff * xdiff + ydiff * ydiff == 1) && // (x^2 + y^2) == 1
               (part_info(next_index).id == part_info(part_index).id) &&
-              (parts[next_index].open == opening ? 0 : 1)) {
+              (parts[next_index].open == 1 && opening)) {
```

Comment:
```
parts[next_index].open == opening ? 0 : 1

int a = parts[next_index].open
bool b = opening

(a == b) ? 0 : 1 <==> (a && b) ? 0 : 1

parts[next_index].open == 1 && opening

(a == 1) && b <==> (a && b)
```

So, yes, sorry the condition is backwards; it should be

```cpp
!(parts[next_index].open  && opening)
```
Good catch; was a careless change on my part.